bokehjs/src/lib/models/math_text/math_text.ts

```diff
@@ -0,0 +1,226 @@
+import {Context2d} from "core/util/canvas"
+import {ImageLoader} from "core/util/image"
+import {CanvasImage} from "models/glyphs/image_url"
+import {Model} from "../../model"
+import {color2css} from "core/util/color"
+import {Size} from "core/types"
+import {View} from "core/view"
+
+type Position = {
+  sx: number
+  sy: number
+  x_anchor?: number | "left" | "center" | "right"
+  y_anchor?: number | "top"  | "center" | "baseline" | "bottom"
+}
+
+/**
+ * Helper class to rendering MathText into Canvas
+ */
+export class MathTextView extends View {
+  override model: MathText
+
+  angle?: number
+  font_size_scale: number = 1.0
+  font: string
+  height: number
+  position: Position = {sx: 0, sy: 0}
+  width: number
+  color: string
+  svg_image: CanvasImage
+  image_is_loading = false
+
+  set visuals(v: visuals.Text) {
+    const color = v.text_color.get_value()
+    const alpha = v.text_alpha.get_value()
+    const style = v.text_font_style.get_value()
+    let size = v.text_font_size.get_value()
+    const face = v.text_font.get_value()
+
+    const {font_size_scale} = this
+    if (font_size_scale != 1.0) {
+      const match = size.match(/^\s*(\d+(\.\d+)?)(\w+)\s*$/)
+      if (match != null) {
+        const [, value,, unit] = match
+        const number = Number(value)
+        if (!isNaN(number))
+          size = `${number*font_size_scale}${unit}`
+      }
+    }
+
+    const font = `${style} ${size} ${face}`
+    this.font = font
+    this.color = color2css(color, alpha)
+  }
+
+  /**
+   * Calculates position of element after considering
+   * anchor and dimensions
+   */
+  protected _computed_position(): {x: number, y: number} {
+    const {width, height} = this
+    const {sx, sy, x_anchor="left", y_anchor="center"} = this.position
+
+    const x = sx - (() => {
+      if (isNumber(x_anchor))
+        return x_anchor*width
+      else {
+        switch (x_anchor) {
+          case "left": return 0
+          case "center": return 0.5*width
+          case "right": return width
+        }
+      }
+    })()
+
+    const y = sy - (() => {
+      if (isNumber(y_anchor))
+        return y_anchor*height
+      else {
+        switch (y_anchor) {
+          case "top": return 0
+          case "center": return 0.5*height
+          case "bottom": return height
+          case "baseline": return 0.5*height
+        }
+      }
+    })()
+
+    return {x, y}
+  }
+
+  /**
+   * Uses the width, height and given angle to calculate the size
+  */
+  size(): Size {
+    const {width, height} = this.getDimensions()
+    const {angle} = this
+
+    if (!angle)
+      return {width, height}
+    else {
+      const c = Math.cos(Math.abs(angle))
+      const s = Math.sin(Math.abs(angle))
+
+      return {
+        width: Math.abs(width*c + height*s),
+        height: Math.abs(width*s + height*c),
+      }
+    }
+  }
+
+  /**
+   * Get width and height from SVGElement
+   */
+  private getDimensions(): Size {
```

Comment:
We use snake case for variable and member names. If you see it otherwise in the code base, there are two reasons for this:

1. It's a historical artifact and it's like this for backwards compatibility reason.
2. It comes from the standard library or external dependencies.